modules/siddhi-core/src/main/java/org/wso2/siddhi/core/util/statistics/metrics/SiddhiMetricsFactory.java

```diff
@@ -33,25 +33,35 @@
  */
 public class SiddhiMetricsFactory implements StatisticsTrackerFactory {
 
-    public LatencyTracker createLatencyTracker(String name, StatisticsManager statisticsManager) {
+    public LatencyTracker createLatencyTracker(String name, StatisticsManager statisticsManager,
+                                               MetricsLogLevel metricsLogLevel) {
         return new SiddhiLatencyMetric(name, ((SiddhiStatisticsManager) statisticsManager).getRegistry());
     }
 
-    public ThroughputTracker createThroughputTracker(String name, StatisticsManager statisticsManager) {
+    public ThroughputTracker createThroughputTracker(String name, StatisticsManager statisticsManager,
+                                                     MetricsLogLevel metricsLogLevel) {
         return new SiddhiThroughputMetric(name, ((SiddhiStatisticsManager) statisticsManager).getRegistry());
     }
 
-    public BufferedEventsTracker createBufferSizeTracker(StatisticsManager statisticsManager) {
+    public BufferedEventsTracker createBufferSizeTracker(StatisticsManager statisticsManager,
+                                                         MetricsLogLevel metricsLogLevel) {
         return new SiddhiBufferedEventsMetric(((SiddhiStatisticsManager) statisticsManager).getRegistry());
     }
 
-    public MemoryUsageTracker createMemoryUsageTracker(StatisticsManager statisticsManager) {
+    public MemoryUsageTracker createMemoryUsageTracker(StatisticsManager statisticsManager,
+                                                       MetricsLogLevel metricsLogLevel) {
         return new SiddhiMemoryUsageMetric(((SiddhiStatisticsManager) statisticsManager).getRegistry());
     }
 
     @Override
-    public StatisticsManager createStatisticsManager(String prefix, String siddhiAppName, List<Element> elements) {
+    public StatisticsManager createStatisticsManager(String prefix, String siddhiAppName, List<Element> elements,
+                                                     boolean isStatisticsEnabled) {
```

Comment:
why isStatisticsEnabled not used ?

Reply:
@suhothayan Added in-order to get isStatsEnabled value to here[1] as per the previous implementation. In siddhi, the value will be taken if elements.size()>0 (these elemets are Siddhi App elements with regards to statistics section. And its taken from the siddhiAppContext.

When it comes to carbon analytics metrices (HA event syncing, theres no siddhi app context right?
But due to the inconsistency, I will move the isStatEnabled in SPStatisticsManager[1] to StreamProcessorDataHolder and use that.

1. https://github.com/wso2/carbon-analytics/blob/0a3db8a1848a276524e3f689f117e4eeae76a4fd/components/org.wso2.carbon.sp.metrics.core/src/main/java/org/wso2/carbon/sp/metrics/core/internal/SPStatisticsManager.java#L33